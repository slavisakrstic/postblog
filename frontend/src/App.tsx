import { ThemeProvider } from "@material-ui/styles";
import { QueryClientProvider } from "react-query";
import { I18nextProvider } from "react-i18next";
import { BrowserRouter } from "react-router-dom";
import { CssBaseline } from "@material-ui/core";

import { queryClient } from "./services/queryClient"

import i18n from "./core/i18n";
import theme from "./theme/createTheme";
import Main from "./Main/Main";

function App() {
  return (
    <BrowserRouter>
      <ThemeProvider theme={theme}>
        <QueryClientProvider client={queryClient}>
          <I18nextProvider i18n={i18n}>
            <CssBaseline />
            <Main />
          </I18nextProvider>
        </QueryClientProvider>
      </ThemeProvider>
    </BrowserRouter>
  );
}

export default App;
