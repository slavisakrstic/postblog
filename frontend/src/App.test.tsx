import renderer from "react-test-renderer";
import { I18nextProvider } from "react-i18next";
import App from "./App";
import i18n from "./core/i18n";

test("should show App component", () => {
  const component = renderer.create(
    <I18nextProvider i18n={i18n}>
      <App />
    </I18nextProvider>
  );
  const componentTree = component.toJSON()
  expect(componentTree).toMatchSnapshot()
});
