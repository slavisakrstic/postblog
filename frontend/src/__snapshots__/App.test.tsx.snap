// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show App component 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthSm"
>
  <div
    className="MuiBox-root MuiBox-root-9"
  >
    <h1
      className="MuiTypography-root makeStyles-loader-8 MuiTypography-h4 MuiTypography-gutterBottom"
    >
      <div
        className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
        data-testid="linear-progress-id"
        role="progressbar"
      >
        <div
          className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
          style={Object {}}
        />
        <div
          className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
          style={Object {}}
        />
      </div>
    </h1>
  </div>
</div>
`;
